showcase/src/main/kotlin/com/recco/showcase/main/MainViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.recco.showcase.main
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.recco.api.model.ReccoFont
+import com.recco.api.model.ReccoPalette
+import com.recco.showcase.ShowcaseRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.WhileSubscribed
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.time.Duration.Companion.seconds
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private val repository: ShowcaseRepository
+) : ViewModel() {
+    private val _viewState = MutableStateFlow(
+        MainUI(
+            selectedPalette = repository.getSelectedReccoPalette(),
+            selectedFont = repository.getSelectedReccoFont()
+        )
+    )
+
+    val viewState: StateFlow<MainUI> = _viewState.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(stopTimeout = 5.seconds),
+        initialValue = MainUI()
+    )
+
+    val selectedReccoPalette get() = repository.getSelectedReccoPalette()
+
+    val selectedReccoFont get() = repository.getSelectedReccoFont()
```

Comment:
[nit] in these cases where we use the `get()` I like to explicitly specify the type of the variable, to avoid goinh to the repository to see it

Reply:
Done 👍